src/tasks/block.rs

```diff
@@ -155,14 +160,24 @@ impl BlockBuilder {
         }
     }
 
-    async fn _get_bundles(&mut self, in_progress: &mut InProgressBlock) {
+    /// Fetches bundles from the cache and ingests them into the in progress block
+    async fn get_bundles(&mut self, host_provider: &Provider, in_progress: &mut InProgressBlock) {
+        // Authenticate before fetching to ensure access to a valid token
+        if let Err(err) = self.bundle_poller.authenticator.authenticate().await {
```

Comment:
if this happens, do we want to continue building the block with regular transactions (as we're doing here) or do we want to fail more aggressively?

Reply:
Can't use the `?` sugar here because it doesn't return a Result type, following the convention of the other functions.